### PR TITLE
CD-64: Mysql driver - add support for username and password

### PR DESCRIPTION
**Pull request by mohamedkallel**

### CODE REVIEW

File: examples/ssh/mysql.js [hunk 1]
```diff
@@ -29,53 +27,45 @@
...
 
 /**
  * @remote_procedure
  * @label Validate Association
  * @documentation This procedure is used to validate if the driver can be applied on a device during association as well as validate any credentials provided
  */
-
 function validate() {
     console.info("Verifying device can respond correctly to command ... ");
     D.device.sendSSHCommand(sshConfig, function(output, error){
```
Comment (mohamedkallel) [inline examples/ssh/mysql.js:64]: use promises for ssh calls

File: examples/ssh/mysql.js [hunk 2]
```diff
@@ -37,45 +78,21 @@ function validate() {
...
-                console.error(error);
-                D.failure();
-            } else {
-                var data = parse(output);
-                reportData(data);
-                if (!isRunning(data)) {
-                    tryStart();
-                }
-            }
+    D.device.sendSSHCommand(sshConfig,function (output, error) {
```
Comment (mohamedkallel) [inline examples/ssh/mysql.js:81]: dupplicate call for the same config, create a function for the ssh call that uses promises

File: examples/ssh/mysql.js [hunk 3]
```diff
@@ -37,45 +78,21 @@ function validate() {
...
+        if (error) {
+            console.error(error);
```
Comment (mohamedkallel) [inline examples/ssh/mysql.js:83]: use checkSshError

File: examples/ssh/mysql.js [hunk 4]
```diff
@@ -37,45 +78,21 @@ function validate() {
...
-            timeout: 10000
-        }, function () {
-            if (error) {
-                console.error(error);
-            }
-        });
-}
-
 function reportData(data) {
+    var _var = D.device.createVariable;
```
Comment (mohamedkallel) [inline examples/ssh/mysql.js:94]: use D.createVariable directly

File: examples/ssh/mysql.js [hunk 5]
```diff
@@ -37,80 +95,52 @@ function validate() {
...
-        {
-            command: "service mysql start",
-            username: D.device.username(),
-            password: D.device.password(),
-            timeout: 10000
-        }, function () {
-            if (error) {
-                console.error(error);
-            }
+    executeCommand(command)
```
Comment (mohamedkallel) [inline examples/ssh/mysql.js:98]: executeCommand(command)
    .then(parse)
    .then(reportData)
    .catch(checkSshError)

**Review by GeorgiTsatsev — APPROVED** [review]

**Pull request merged**